### DISCUSSION TITLE
How do I use the API to post a new record with a many-to-many / linked record relationship?

### DISCUSSION THREAD

**Discussion by 684efs3** [nocodb/nocodb#3880] [Q&A]
I am using Python and have been successful in uploading a new record. However, I want to upload a record that has a linked record. How do I do this? 

When I use TablePlus (or something similar) to view the tables in my Postgres, I do not see a foreign key in the table. Is this in one of the NocoDB generated table? (I see a lot of _nc_m2m_... tables. I assume m2m means many-to-many). Any help would be much appreciated!

**Comment by 684efs3** [comment]
This is a many-to-many relationship.

**Comment by pedralmeida22** [comment]
### Documentation 
https://apis.nocodb.com/#tag/DB-Table-Row/operation/db-table-row-nested-add


I was able to achieve upload a new record with a many-to-many relation after a few steps:
1. Upload a new record
2. Find the row IDs in both tables to link (I used the GET _find-one_ method)
3. Update record with link


### Python example:
Consider two tables: TableA and TableB.
The following code will add a record (_new_record_) to TableB, retrieve the IDs from the records to link in both tables and, finally, update the _new_record_ with the link.

```
import requests

access_token = "<your_access_token>"

s = requests.Session()

# Create row
create_row_request = requests.post('http://localhost:8080/api/v1/db/data/v1/ORG/TableB',
                  headers={'Content-Type': 'application/json',
                           'xc-token': access_token,
                           },
                  json={
                      "Title": "titleB"
                  }
                  )

# Find row
find_one_request = requests.get('http://localhost:8080/api/v1/db/data/v1/ORG/TableB/find-one?where=(Title,eq,titleB)',
                                headers={'Content-Type': 'application/json',
                                         'xc-token': access_token,
                                         }
                                )
row_id = find_one_request.json()["Id"]

# Get row id (other table)
find_one_request2 = requests.get('http://localhost:8080/api/v1/db/data/v1/ORG/TableA/find-one?where=(Title,eq,titleA)',
                                 headers={'Content-Type': 'application/json',
                                          'xc-token': access_token,
                                          }
                                 )
row_id_other_table = find_one_request2.json()["Id"]


# Link row
link_row = requests.post(f'http://localhost:8080/api/v1/db/data/v1/ORG/TableB/{row_id}/mm/columnName/{row_id_other_table}',
                         headers={'Content-Type': 'application/json',
                                  'xc-token': access_token,
                                  }
                         )
print(f"Status Code: {link_row.status_code}, Response: {link_row.json()}")
```

Output:

![image](https://github.com/nocodb/nocodb/assets/40249872/0a0940a1-04ea-4c9e-bf27-aaff3a75298c)
